qt-bugs@ issue : N180629
Trolltech task ID : none
bugs.kde.org number : None
applied: yes
author: Rafael Fernández López <ereslibre@kde.org>

This patch fixes a division by zero that happens on the QTreeView widget
if the ScrollPerPixel scroll mode is set.

--- a/src/gui/itemviews/qtreeview.cpp
+++ b/src/gui/itemviews/qtreeview.cpp
@@ -2983,6 +2983,9 @@
     }
     // ScrollMode == ScrollPerPixel
     if (uniformRowHeights) {
+        if (!defaultItemHeight)
+            return -1;
+
         if (offset)
             *offset = -(value % defaultItemHeight);
         return value / defaultItemHeight;
